src/StableHashTraits.jl

```diff
@@ -425,16 +425,59 @@ function validate_name(str)
 end
 
 qname_(T, name) = validate_name(cleanup_name(string(parentmodule(T), '.', name(T))))
+
 qualified_name_(fn::Function) = qname_(fn, nameof)
-qualified_type_(fn::Function) = qname_(fn, string)
+qualified_type1_(fn::Function) = qname_(fn, string)
 qualified_name_(x::T) where {T} = qname_(T <: DataType ? x : T, nameof)
-qualified_type_(x::T) where {T} = qname_(T <: DataType ? x : T, string)
+qualified_type1_(x::T) where {T} = qname_(T <: DataType ? x : T, string)
 qualified_(T, ::Val{:name}) = qualified_name_(T)
-qualified_(T, ::Val{:type}) = qualified_type_(T)
+qualified_(T, ::Val{:type1}) = qualified_type1_(T)
+qualified_(T, ::Val{:type2}) = qualified_type2_(T)
+
+"""
```

Comment:
This function was based of the discussion in #41. The older implementation just replaced "workspace#" strings, but now I can properly make a check rather than just assuming any use of "workspace#" means it's a Pluto variable/module.